pkg/reconciler/route/resources/service.go

```diff
@@ -126,16 +126,13 @@ func makeK8sService(ctx context.Context, route *v1alpha1.Route, targetName strin
 	}, nil
 }
 
-func makeServiceSpec(ingress netv1alpha1.IngressAccessor, isPrivate bool) (*corev1.ServiceSpec, error) {
+func makeServiceSpec(ingress netv1alpha1.IngressAccessor) (*corev1.ServiceSpec, error) {
 	ingressStatus := ingress.GetStatus()
 
-	var lbStatus *netv1alpha1.LoadBalancerStatus
-
-	if isPrivate {
-		lbStatus = ingressStatus.PrivateLoadBalancer
-	} else {
-		lbStatus = ingressStatus.PublicLoadBalancer
-	}
+	// Always use private load balancer here,
+	// because k8s service is only useful for inter-cluster communication.
+	// External communication will be handle via ingress gateway, which won't be affected by what is configured here.
+	lbStatus := ingressStatus.PrivateLoadBalancer
```

Comment:
Similarly here. the cluster local ingress gateway is not a requirement. We should use it when it exists. But if it does not exist, we fall back to public ingress.